src/NHibernate/Linq/Functions/StringGenerator.cs

```diff
@@ -210,18 +216,28 @@ public override HqlTreeNode BuildHql(MethodInfo method, Expression targetObject,
 
 	public class IndexOfGenerator : BaseHqlGeneratorForMethod
 	{
+		private static readonly MethodInfo MethodWithComparer1 = ReflectHelper.GetMethodDefinition<string>(x => x.IndexOf(string.Empty, default(StringComparison)));
+		private static readonly MethodInfo MethodWithComparer2 = ReflectHelper.GetMethodDefinition<string>(x => x.IndexOf(string.Empty, 0, default(StringComparison)));
+
 		public IndexOfGenerator()
 		{
 			SupportedMethods = new[]
 									{
 										ReflectHelper.GetMethodDefinition<string>(s => s.IndexOf(' ')),
 										ReflectHelper.GetMethodDefinition<string>(s => s.IndexOf(" ")),
 										ReflectHelper.GetMethodDefinition<string>(s => s.IndexOf(' ', 0)),
-										ReflectHelper.GetMethodDefinition<string>(s => s.IndexOf(" ", 0))
+										ReflectHelper.GetMethodDefinition<string>(s => s.IndexOf(" ", 0)),
+										MethodWithComparer1,
+										MethodWithComparer2,
 									};
 		}
 		public override HqlTreeNode BuildHql(MethodInfo method, Expression targetObject, ReadOnlyCollection<Expression> arguments, HqlTreeBuilder treeBuilder, IHqlExpressionVisitor visitor)
 		{
+			if (LogIgnoredStringComparisonParameter(method, MethodWithComparer1, MethodWithComparer2))
```

Comment:
Easier to check that function has 3 arguments.

Reply:
Not in case ambiguity. Like here 2 arguments for "s.IndexOf(" ", 0)" and `s.IndexOf("", StringComparison.Oridnal)`

And what you mean be easier? This method is easier to reuse in multiple places. And argument count is error prone approach when doing changes in multiple places?